x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rules_list.tsx

```diff
@@ -345,6 +356,24 @@ export const RulesList: React.FunctionComponent = () => {
     }
   }
 
+  async function loadRuleTags() {
```

Comment:
I want to give more context to this function here. So there's an awkward behaviour when I added the tag filters to the aggregation, because we use 1 aggregation function to get a list of tags and all of the rule execution status values, if we started filtering down the aggregation with tag filters, the list of tags would also get filtered, and thus it would replace the list of available tags with a smaller subset of tags (which is incorrect).

To fix this, i considered only calling `setState` on the tags list if the tagsFilter is empty. But this is also incorrect because we miss the edge case where the user adds new tags or a new rule while the tag filter is applied. 

From there, I decided to add this function, which calls the aggregation with empty params, as a part of the data fetch. I think the downside here is we're doing the aggregation twice now, which has performance implications.

But aside from doing something client side like merging in new tags in react, we might eventually want to have a separate aggregation just to get a list of tags. Let me know if we want to do that in this PR or I could do it in another PR.

Reply:
Is it possible to understand this more? How can I simulate this? Or can I see this happening on an older version of this PR?

Reply:
Hmm 

What about something like:
```
 async function loadRuleAggs() {
    try {
      const rulesAggs = await loadRuleAggregations({
        http,
        searchText,
        typesFilter,
        actionTypesFilter,
        ruleStatusesFilter,
        tagsFilter,
      });
      if (rulesAggs?.ruleExecutionStatus) {
        setRulesStatusesTotal(rulesAggs.ruleExecutionStatus);
      }
      if (rulesAggs?.ruleTags && rulesAggs?.ruleTags.length > tags.length) {
        setTags(rulesAggs.ruleTags);
      }
    } 
```

I guess this doesn't work if we delete a rule and a tag is only associated with that rule though

I think we should separate these two things out into separate "things" - api that is called that is assumed to return _filtered_ data and a separate api call that is expected to return _total_ data. We might need to add something to the `lib/rule_api` layer too but it's too confusing to keep this logic as-is - definitely recommend some delineation here to make it clear in the future.

Reply:
yea thought so, ok I'll add a new endpoint to expose tags